monai/inferers/merger.py

```diff
@@ -246,8 +247,20 @@ def __init__(
         self.value_dtype = value_dtype
         self.count_dtype = count_dtype
         self.store = store
-        self.value_store = zarr.storage.TempStore() if value_store is None else value_store
-        self.count_store = zarr.storage.TempStore() if count_store is None else count_store
+        if version_geq(get_package_version("zarr"), "3.0.0"):
+            if value_store is None:
+                tmpdir = TemporaryDirectory()
+                self.value_store = zarr.storage.LocalStore(tmpdir.name)
+            else:
+                self.value_store = value_store
+            if count_store is None:
+                tmpdir = TemporaryDirectory()
+                self.count_store = zarr.storage.LocalStore(tmpdir.name)
+            else:
+                self.count_store = count_store
+        else:
```

Comment:
```suggestion
        if version_geq(get_package_version("zarr"), "3.0.0"):
            if value_store is None:
                self.tmpdir = TemporaryDirectory()
                self.value_store = zarr.storage.LocalStore(self.tmpdir.name)
            else:
                self.value_store = value_store
            if count_store is None:
                self.tmpdir = TemporaryDirectory()
                self.count_store = zarr.storage.LocalStore(self.tmpdir.name)
            else:
                self.count_store = count_store
        else:
            self.tmpdir = None
```

What I meant was that the lifecycle of the tmpdir object should follow that of the containing object since this will be the same as that for the storage objects.